### PR TITLE
Bump version to 2.21.1

### PR DESCRIPTION
**Pull request by matthewfala**
Signed-off-by: Matthew Fala <falamatt@amazon.com>

This build relies on a branch on my fork which contains [1.8.9](https://github.com/fluent/fluent-bit/releases/tag/v1.8.9) + the same commits we cherry-picked into the last release for the TLS and mbedtls fixes: https://github.com/matthewfala/fluent-bit/tree/1_8_9_patch

```
# Apply Fluent Bit TLS and mbedtls Patch
git fetch https://github.com/krispraws/fluent-bit.git tls_net_read_fix \
  && git cherry-pick 8d1cfeb5ba830b360fe6e1228190ed900842a3ea; \
git fetch https://github.com/zhonghui12/fluent-bit.git custom-1.8.7 \
  && git cherry-pick 30fc6305695623cbc95d51df07ae185dfec8bff2
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AWS_FOR_FLUENT_BIT_VERSION [hunk 1]
```diff
@@ -1 +1 @@
-2.21.0
\ No newline at end of file
+2.21.1
```
Comment (matthewfala) [inline AWS_FOR_FLUENT_BIT_VERSION:1]: Changed to 2.21.1

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,18 @@
 # Changelog
 
+### 2.21.1
```
Comment (matthewfala) [inline CHANGELOG.md:3]: Changed to 2.21.1

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+This release includes:
+* An Amazon Linux 2 Base
+* Fluent Bit [1.8.9](https://fluentbit.io/announcements/v1.8.9/)
+* Amazon CloudWatch Logs for Fluent Bit 1.6.3
+* Amazon Kinesis Streams for Fluent Bit 1.8.0
+* Amazon Kinesis Firehose for Fluent Bit 1.6.1
+
+Compared to `2.21.0` this release adds:
+* Bug - Avoid double free for multiline buffer [fluentbit:4243](https://github.com/fluent/fluent-bit/pull/4243)
```
Comment (matthewfala) [inline CHANGELOG.md:12]: Added bug fix report

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,18 @@
...
 
+### 2.21.1
+This release includes:
+* An Amazon Linux 2 Base
+* Fluent Bit [1.8.9](https://fluentbit.io/announcements/v1.8.9/)
+* Amazon CloudWatch Logs for Fluent Bit 1.6.3
+* Amazon Kinesis Streams for Fluent Bit 1.8.0
+* Amazon Kinesis Firehose for Fluent Bit 1.6.1
+
+Compared to `2.21.0` this release adds:
```
Comment (matthewfala) [inline CHANGELOG.md:11]: Kept "Compared to..." since we have an actual bug fix from @hossain-rayhan

**Review by matthewfala — COMMENTED** [review]
Revision diff
```
$ git diff
diff --git a/AWS_FOR_FLUENT_BIT_VERSION b/AWS_FOR_FLUENT_BIT_VERSION
index f1270b4..bd9c559 100644
--- a/AWS_FOR_FLUENT_BIT_VERSION
+++ b/AWS_FOR_FLUENT_BIT_VERSION
@@ -1 +1 @@
-2.22.0
\ No newline at end of file
+2.21.1
\ No newline at end of file
diff --git a/CHANGELOG.md b/CHANGELOG.md
index 4d7dc1f..ed93fe2 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,6 @@
 # Changelog
 
-### 2.22.0
+### 2.21.1
 This release includes:
 * An Amazon Linux 2 Base
 * Fluent Bit [1.8.9](https://fluentbit.io/announcements/v1.8.9/)
@@ -9,6 +9,7 @@ This release includes:
 * Amazon Kinesis Firehose for Fluent Bit 1.6.1
 
 Compared to `2.21.0` this release adds:
+* Bug - Avoid double free for multiline buffer [fluentbit:4243](https://github.com/fluent/fluent-bit/pull/4243)
 * Bug - Fix return value from `tls_net_read` [fluentbit:4098](https://github.com/fluent/fluent-bit/issues/4098)
 * Bug - Downgrade `mbedtls` to 2.24.0 to fix the performance regression issue in `mbedtls` 2.26.0 [fluentbit:4110](https://github.com/fluent/fluent-bit/issues/4110)
 ```

File: CHANGELOG.md (hunk 4 above)
Comment (PettitWesley) [reply CHANGELOG.md:11]: yea but the compared too isn't true for the other ones. So Its still a lie. I think we should change the language for the 2 fixes we keep adding in.

Comment (matthewfala) [reply CHANGELOG.md:11]: Oh okay. I see. Will fix.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,21 @@
...
+* Bug - Avoid double free for multiline msgpack buffer [fluentbit:4243](https://github.com/fluent/fluent-bit/pull/4243)
+* Bug - On Multiline handling, use file inode as stream_id [fluentbit:3886](https://github.com/fluent/fluent-bit/issues/3886)
```
Comment (matthewfala) [inline CHANGELOG.md:13]: Add Multiline handling bug fix call out

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,21 @@
...
+
+Same as `2.21.0`, this release includes the following fixes for AWS customers that we are working on getting accepted upstream:
```
Comment (matthewfala) [inline CHANGELOG.md:15]: Alter format of AWS image specific patch callout

File: CHANGELOG.md (hunk 5 above)
Comment (matthewfala) [inline CHANGELOG.md:13]: Add Multiline handling bug fix callout

**Review by matthewfala — COMMENTED** [review]
Second diff
```
diff --git a/CHANGELOG.md b/CHANGELOG.md
index ed93fe2..e71ab37 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,7 +9,10 @@ This release includes:
 * Amazon Kinesis Firehose for Fluent Bit 1.6.1
 
 Compared to `2.21.0` this release adds:
-* Bug - Avoid double free for multiline buffer [fluentbit:4243](https://github.com/fluent/fluent-bit/pull/4243)
+* Bug - Avoid double free for multiline msgpack buffer [fluentbit:4243](https://github.com/fluent/fluent-bit/pull/4243)
+* Bug - On Multiline handling, use file inode as stream_id [fluentbit:3886](https://github.com/fluent/fluent-bit/issues/3886)
+
+Same as `2.21.0`, this release includes the following fixes for AWS customers that we are working on getting accepted upstream:
 * Bug - Fix return value from `tls_net_read` [fluentbit:4098](https://github.com/fluent/fluent-bit/issues/4098)
 * Bug - Downgrade `mbedtls` to 2.24.0 to fix the performance regression issue in `mbedtls` 2.26.0 [fluentbit:4110](https://github.com/fluent/fluent-bit/issues/4110)
```

**Review by PettitWesley — APPROVED** [review]

**Comment by zhonghui12** [comment]
The commit message needs to be updated to `Bump version to 2.21.1` as well

**Pull request merged**